CMakeLists.txt

```diff
@@ -30,160 +30,136 @@ set(IRODS_MIN_VERSION "4.2.0")
-              message(STATUS "Setting unspecified CPACK_GENERATOR to ${CPACK_GENERATOR}. This is the correct setting for normal builds.")
-            endif()
-            
-            include(CPack)
-
-        elseif (DEFINED ENV{IRODS_40_COMPAT})
-            set(irods_link_obj_path 
-                    "$ENV{IRODS_PATH}/lib/irods/libirods_client.so"
-                    "$ENV{IRODS_PATH}/lib/irods/libirods_client_api.so"
-                    "-lrt"
-                    "-lcurl"
-                    "-lpthread"
-                    "-pthread"
-                    "$ENV{IRODS_PATH}/lib/irods/libjansson.a"
-                    "$ENV{IRODS_PATH}/lib/irods/libboost_filesystem.a"
-                    "$ENV{IRODS_PATH}/lib/irods/libboost_regex.a"
-                    "$ENV{IRODS_PATH}/lib/irods/libboost_system.a"
-                    "$ENV{IRODS_PATH}/lib/irods/libboost_thread.a")
-        else()
-            set(irods_link_obj_path 
-                    "$ENV{IRODS_PATH}/lib/libirods_client.so"
-                    "$ENV{IRODS_PATH}/lib/libirods_client_api.so"
-                    "-lrt"
-                    "-lpthread"
-                    "-pthread"
-                    "-lssl"
-                    "-lcurl"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libjansson.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_chrono.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_filesystem.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_program_options.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_regex.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_system.a"
-                    "$ENV{IRODS_PATH}/lib/irods/externals/libboost_thread.a")
-        endif()
-    else()
-        set(irods_include_path_list "$ENV{IRODS_PATH}/lib/api/include"
-                "$ENV{IRODS_PATH}/lib/core/include"
-                "$ENV{IRODS_PATH}/server/icat/include"
-                "$ENV{IRODS_PATH}/lib/md5/include/"
-                "$ENV{IRODS_PATH}/lib/sha1/include/"
-                "$ENV{IRODS_PATH}/server/core/include/"
-                "$ENV{IRODS_PATH}/server/drivers/include/"
-                "$ENV{IRODS_PATH}/server/re/include/")
-        set(irods_link_obj_path "$ENV{IRODS_PATH}/lib/core/obj/libRodsAPIs.a")
-    endif()
-else()
-    message( FATAL_ERROR "IRODS_PATH is NOT defined, CMake will exit." )
+add_definitions(-DIRODS_HEADER_HPP)
+find_package(IRODS REQUIRED CONFIG)
+
+set(CMAKE_INSTALL_RPATH ${IRODS_EXTERNALS_FULLPATH_CLANG_RUNTIME}/lib)
+set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -stdlib=libc++")
+set(CMAKE_MODULE_LINKER_FLAGS "${CMAKE_MODULE_LINKER_FLAGS} -stdlib=libc++")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -stdlib=libc++")
+add_compile_options(-nostdinc++ -O0 -Wno-deprecated-register)
```

Comment:
Instead of manually adding the `-stdlibc++` and `-nostdinc++` flags, manually linking to `libc++abi`, and manually adding `${IRODS_EXTERNALS_FULLPATH_CLANG}/include/c++/v1` to the include path, use the new `UseLibCXX` module. An example of how this is done can be seen in irods/irods_client_icommands#192

Reply:
I think we're planning to initially release this against 4.2.8, is that right, Justin?

If so, we can't use the new `UseLibCXX` yet.

Reply:
I don't know what the plan is.

Reply:
Justin says we *do* need 4.2.9 - so we can use all the nice things in 4-2-stable in this CMakeLists.txt file.